packages/thirdweb/src/transaction/serialize-transaction.ts

```diff
@@ -48,243 +33,28 @@ export type SerializeTransactionOptions<
-
-  let serializedTransaction = [
-    nonce ? toHex(nonce) : "0x",
-    gasPrice ? toHex(gasPrice) : "0x",
-    gas ? toHex(gas) : "0x",
-    (to ?? "0x") as Hex,
-    value ? toHex(value) : "0x",
-    data ?? "0x",
-  ];
-
-  if (typeof v !== "undefined") {
-    const v_ = (() => {
-      // EIP-155 (inferred chainId)
-      if (v >= 35n) {
-        const inferredChainId = (v - 35n) / 2n;
-        if (inferredChainId > 0) return v;
-        return 27n + (v === 35n ? 0n : 1n);
-      }
-
-      // EIP-155 (explicit chainId)
-      if (chainId > 0) return BigInt(chainId * 2) + BigInt(35n + v - 27n);
-
-      // Pre-EIP-155 (no chainId)
-      const v_ = 27n + (v === 27n ? 0n : 1n);
-      if (v !== v_) throw new Error(`Invalid legacy signature value v: ${v}`);
-      return v_;
-    })();
-
-    if (typeof r === "undefined")
-      throw new Error(`Invalid legacy signature value r: ${r}`);
-    if (typeof s === "undefined")
-      throw new Error(`Invalid legacy signature value s: ${s}`);
-
-    serializedTransaction = [
-      ...serializedTransaction,
-      toHex(v_),
-      r === "0x00" ? "0x" : r,
-      s === "0x00" ? "0x" : s,
-    ];
-  } else if (chainId > 0) {
-    serializedTransaction = [
-      ...serializedTransaction,
-      toHex(chainId),
-      "0x",
-      "0x",
-    ];
-  }
-
-  return toRlp(serializedTransaction) as TransactionSerializedLegacy;
-}
-
-export function toYParitySignatureArray(
-  signature: ExactPartial<Signature>,
-): Hex[] {
-  const { v, yParity } = signature;
-
-  if (typeof signature.r === "undefined") return [];
-  if (typeof signature.s === "undefined") return [];
-
-  if (typeof v === "undefined" && typeof yParity === "undefined") return [];
-
-  const r = trim(signature.r);
-  const s = trim(signature.s);
-
-  const yParity_ = (() => {
-    if (yParity === 1 || yParity === 0) return yParity ? toHex(1) : "0x";
-    if (typeof yParity !== "undefined")
-      throw new Error(`Invalid signature value for yParity: ${yParity}`);
-
-    if (v === 0n || v === 27n) return "0x";
-    if (v === 1n || v === 28n) return toHex(1);
-    throw new Error(`Invalid signature value v: ${v}`);
+  // This is to maintain compatibility with our old interface (including the signature in the transaction object)
+  const signature = (() => {
+    if (options.signature) return options.signature;
+    if (transaction.v === undefined && transaction.yParity === undefined) {
+      return undefined;
+    }
+
+    if (transaction.r === undefined || transaction.s === undefined) {
+      throw new Error("Invalid signature provided with transaction");
+    }
+
+    return {
+      v: transaction.v,
+      r: transaction.r,
+      s: transaction.s,
+      yParity: transaction.yParity,
+    };
   })();
-  return [yParity_, r === "0x00" ? "0x" : r, s === "0x00" ? "0x" : s];
+
+  return _serializeTransaction(transaction, signature as Signature); // Trust the options type-checking did its job and that the converted signature mirrors that type
```

Comment:
instead of the `as` here can you define the type on the `const signature: Signature = ...` above - then it matches here and if that complains then ur missing something in there, no?

Reply:
The type error is just because it has certain mutual exclusivity checks between `v` and `yParity`